### ISSUE TITLE
Empty password

### ISSUE THREAD

**Issue by EmanueleMinotto** [elcodi/elcodi#995]
`Elcodi\Component\User\Entity\Abstracts\AbstractUser::setPassword` doesn't allow to set the password to `null` but allows empty strings.

**Comment by TomasVotruba** [comment]
What's expected behavior?

**Comment by mmoreram** [comment]
Password shoul'nt be nor null nor empty, right?

**Comment by EmanueleMinotto** [comment]
exactly, the minimum check should be executing a `trim` on the input

**Comment by TomasVotruba** [comment]
Ok. Exception shall be thrown?

**Comment by EmanueleMinotto** [comment]
yes, imo an `\InvalidArgumentException` should be thrown, do you agree @mmoreram?

**Comment by kodermax** [comment]
Now, we don't have check for current password.
And we don't see anything warning messages.

**Comment by kodermax** [comment]
may be don't use current password, only use two fields new password and repeat password.

**Comment by mmoreram** [comment]
@EmanueleMinotto absolutely :) But after that, this should be managed by the final application as well.
